plugins/modules/packaging/language/npm.py

```diff
@@ -174,7 +174,7 @@ def _exec(self, args, run_in_check_mode=False, check_rc=True):
                 cmd.append('--ignore-scripts')
             if self.unsafe_perm:
                 cmd.append('--unsafe-perm')
-            if self.name:
+            if self.name and 'list' not in cmd:
```

Comment:
Why not have an additional optional parameter `add_package_name=True` which is set to `False` in the invocation inside `list()`?

Reply:
ok.